deepchem/feat/base_classes.py

```diff
@@ -201,6 +203,13 @@ def featurize(self, molecules, log_every_n=1000):
         if isinstance(mol, str):
           # mol must be a SMILES string so parse
           mol = Chem.MolFromSmiles(mol)
+          # TODO (ytz) this is a bandage solution to reorder the atoms
+          # so that they're always in the same canonical order.
+          # Presumably this should be correctly implemented in the
+          # future for graph mols.
+          if mol:
+            new_order = rdmolfiles.CanonicalRankAtoms(mol)
+            mol = rdmolops.RenumberAtoms(mol, new_order)
```

Comment:
I've added these fixes in since this is what the DataLoader classes did to smiles as they were loading them. We should make our featurizers consistent with our usual behavior